build-system/runner/src/org/ampproject/AmpCommandLineRunner.java

```diff
@@ -68,6 +69,8 @@ protected AmpCommandLineRunner(String[] args) {
     }
     CompilerOptions options = super.createOptions();
     options.setCollapseProperties(true);
+    // Warn on improper uses of `this` that `collapseProperties` assumes.
+    options.setCheckGlobalThisLevel(CheckLevel.WARNING);
```

Comment:
Could this instead be added here: https://github.com/ampproject/amphtml/blob/master/build-system/tasks/compile.js#L269

Our config is a bit over the place, but we try to use the flags where available instead of Java.